### PR TITLE
Association aggregation composition

### PR DESCRIPTION
**Pull request by Fourteen98**
### New Features:
- Created a ClassRoom class
- Created has many/belongs-to relationship between ClassRoom and student
- Created a class Book
- Created a class Rental
- Create many-to-many relationship between Person and Book using intermediate class Rental

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (dasileker) [inline book.rb]: - [x] hello the book also should get access to the rental not to read.

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals << rental
+    rental.book = self
+  end
```
Comment (dasileker) [inline book.rb:13]: - [x] The def rental should take two argument examples:(student or teacher) already initialized and age. and then we create a new rental.

File: rental.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+class Rental
+  attr_accessor :date
```
Comment (dasileker) [inline rental.rb]: - [x] also in the rental we should get assess also to the book and person.

File: person.rb [hunk 4]
```diff
@@ -4,7 +4,7 @@
 
 class Person < Nameable
   attr_accessor :name, :age
```
Comment (dasileker) [inline person.rb]: the same here also for the rental should get access not read.

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @Fourteen98,** 

### **Good job so far!**
There are some issues that you still need to work on to go to the next project but you are almost there!

## **Highlights**
-  linters are green.
- PR is very nice.
- commits are consistent.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by omarsalem7 — APPROVED** [review]
Hi @Fourteen98 😊, 
# STATUS: APPROVED! 🥇 💯 

giphy

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 🚀 
Congratulations! 🎉 

## Highlights
✅ Descriptive pull request 
✅ Project has met all the technical aspects you needed for this project. Keep rocking! 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**